backend/src/main/java/de/bund/digitalservice/ris/norms/utils/EidConsistencyGuardian.java

```diff
@@ -108,127 +90,56 @@ private static boolean isReferenceValid(
     return false;
   }
 
-  private static void updateElementEids(
-      final Element element, final String parentEid, Element rootElement) {
-    final Map<String, Integer> childTypeCount =
-        new HashMap<>(); // Track count of child nodes of each type
-    final NodeList childNodes = element.getChildNodes();
-    for (int i = 0; i < childNodes.getLength(); i++) {
-      final Node node = childNodes.item(i);
-      if (node.getNodeType() == Node.ELEMENT_NODE) {
-        final Element childElement = (Element) node;
-
-        // Keep trap of index counting per node type
-        final String childTagName = childElement.getTagName();
-        childTypeCount.put(childTagName, childTypeCount.getOrDefault(childTagName, 0) + 1);
-
-        if (childElement.hasAttribute("eId")) {
-          final String currentEid = childElement.getAttribute("eId");
-          final String correctEid =
-              determineCorrectEid(childTypeCount.get(childTagName), currentEid, parentEid);
-
-          if (!currentEid.equals(correctEid)) {
-            updateReferences(currentEid, correctEid, rootElement);
-            childElement.setAttribute("eId", correctEid);
-            updateElementEids(
-                childElement,
-                correctEid,
-                rootElement); // Recursively update child elements with corrected eId
-          } else {
-            updateElementEids(
-                childElement,
-                currentEid,
-                rootElement); // Recursively update child elements with old eId
-          }
-        } else {
-          updateElementEids(childElement, null, rootElement);
-        }
-      }
-    }
-  }
+  private static void updateElementEids(final Node node, Element rootElement) {
+    final Map<String, String> oldToNewEIdMap = new HashMap<>();
+    List<Node> nodesToUpdate = new ArrayList<>();
+    nodesToUpdate.add(node);
 
-  private static String determineCorrectEid(int currentIndex, String oldEid, String newParentEid) {
-    final StringBuilder newEid = new StringBuilder();
+    while (!nodesToUpdate.isEmpty()) {
+      var currentNode = nodesToUpdate.removeLast();
 
-    String currentElementId;
+      if (currentNode instanceof Element element) {
+        final var expectedEId = EId.forNode(currentNode);
+        final var currentEId = EId.fromNode(currentNode);
 
-    // Check first if eId contains parent eIds
-    if (oldEid.contains("_")) {
-      // 1. ParentEid is not the same
-      final int lastIndex = oldEid.lastIndexOf('_');
+        if (expectedEId != currentEId) {
+          if (expectedEId.isPresent()) {
```

Comment:
What do we do if the expectedEid is not present? Are we throwing an exception in the Eid class if the expectedEid could not be calculated? And also what about is the currentEid is empty? Also throw an exception? Because we could have the case that neither of them are present and then the resulted XML would not be valid right?

Reply:
There are some elements in the xml that do not need an eid so both being empty is totally find. I was wondering what to do if we only have a current eid but no expected one, but decided that it is best to keep the eid in that case to not delete eids from elements we might not support yet

Reply:
Alright. This means that we may have nodes without a corrected eId or even have some with no eId at all altough they should have one right?

Reply:
Got it. Even if a parent node didn't have an eId it will be corrected in the iteration before. Thanks!